superset/views/core.py

```diff
@@ -2391,6 +2396,36 @@ def select_star(self, database_id, table_name, schema=None):
             mydb.select_star(table_name, schema, latest_partition=True, show_cols=True)
         )
 
+    @has_access_api
+    @expose("/estimate_query_cost/<database_id>/", methods=["POST"])
+    @expose("/estimate_query_cost/<database_id>/<schema>/", methods=["POST"])
+    @event_logger.log_this
+    def estimate_query_cost(self, database_id, schema=None):
+        sql = json.loads(request.form.get("sql"))
+        template_params = json.loads(request.form.get("templateParams") or "{}")
+
+        if len(template_params) > 0:
+            return json_error_response(
+                "Query cost estimation does not support template parameters", status=400
```

Comment:
why doesn't it support template parameters? Couldn't we just resolve the templates then run the estimation?

Reply:
The template processing is tightly coupled with the database (for resolving macros related to latest partition, etc.), and it was hard to bring it on here.

Reply:
Actually I got this to work.